packages/react-dev-utils/launchEditor.js

```diff
@@ -11,10 +11,62 @@
 const fs = require('fs');
 const path = require('path');
 const child_process = require('child_process');
+const EventEmitter = require('events').EventEmitter;
 const os = require('os');
 const chalk = require('chalk');
 const shellQuote = require('shell-quote');
 
+// Inspired by https://github.com/rannn505/node-powershell
+const EOI = 'EOI';
+class PowerShell extends EventEmitter {
+  constructor() {
+    super();
+
+    this._proc = child_process.spawn(
+      'powershell.exe',
+      ['-NoLogo', '-NoProfile', '-NoExit', '-Command', '-'],
+      {
+        stdio: 'pipe',
+      }
+    );
+
+    this._proc.on('error', () => {
+      // Needs to be registered... already catched by if-statement below
+    });
+
+    if (!this._proc.pid) {
+      throw new Error('Failed to start PowerShell');
+    }
+
+    // Initialize counters for mapping events
+    this._callbackCounter = 0;
+    this._resolveCounter = 0;
+
+    let output = [];
+    this._proc.stdout.on('data', data => {
+      if (data.indexOf(EOI) !== -1) {
+        const eventName = 'resolve' + ++this._callbackCounter;
+        this.emit(eventName, output.join(''));
+        output = [];
+      } else {
+        output.push(data);
+      }
+    });
+  }
+
+  invoke(cmd) {
+    return new Promise(resolve => {
+      const eventName = 'resolve' + ++this._resolveCounter;
```

Comment:
Can you try without using EventEmitter? I find code built on events (especially with dynamic names) easy to mess up by accidentally triggering or listening to a wrong event. I would prefer plain callback lists.

Reply:
Sure.